### ISSUE TITLE
PNGs getting modified and breaking upon upload to media library

### ISSUE THREAD

**Issue by cooperwfloyd** [strapi/strapi#16227]
### Required System information

- Node.js 16.15.1
- NPM 8.11.0
- Bug introducted on Strapi 4.6.0, still an issue on the latest Strapi 4.8.2. Works fine on Strapi 4.5.6.
- Same issue on both local SQLite and cloud Postgres
- Local OS is MacOS 13.2.1, cloud environment is a render.com Pro Max Web Service
- Is your project Javascript or Typescript: JavaScript
- No custom upload configurations, optimizations, transformations, etc. in place

### Describe the bug

When uploading PNGs to the Strapi media library on and after 4.6.0, the image appears correctly in the preview pane before uploading, but after uploading it appears broken and distorted. This has been the case for all PNGs lately, not just the attached example.

I tried re-exporting and changing the PNG export options and exporting from different applications too, but that ended up having the same effect. Workaround at this point is using a format other than a PNG, but this is not a viable long-term solution.

### Steps to reproduce the behavior

1. Navigate to the media library and select the attached PNG to be uploaded in the environment specified above.
2. Notice that the file will look normal in the preview pane, but after clicking "Upload 1 asset to media library", notice that the image appears distorted and broken in the media library

### Expected behavior

For the image to look the same after it is uploaded as before it is uploaded (works fine on Strapi 4.5.6).

### Screenshots

<img width="1134" alt="Screenshot 2023-03-27 at 1 51 24 PM" src="https://user-images.githubusercontent.com/18223307/228043417-d2b11397-c0c9-439b-8f5e-0c6b10c22cd1.png">

![test](https://user-images.githubusercontent.com/18223307/228044521-16e909fe-f268-48b2-8e6a-398cd2387c91.png)

**Comment by beniaminrychter** [comment]
I occurred the same issue. But the problem is only with some PNG images. Below you can find two icons, and the only difference is that `broken-icon` has transparent elements, but `ok-icon` **has not**.

Broken ❌: <img width="37" alt="broken-icon" src="https://user-images.githubusercontent.com/27733472/228211770-fd6d5744-145d-49db-8435-9ce717568475.png">
OK ✅: <img width="37" alt="ok-icon" src="https://user-images.githubusercontent.com/27733472/228211775-f5311805-90ee-412d-a8da-6cbc90ecebb7.png">

**Comment by cooperwfloyd** [comment]
quick update here since a new @strapi/strapi version was released — installed 4.9.0 this morning to test and I'm still seeing the same issue as above

**Comment by JWandscheer** [comment]
This could be related to a bug in dependency package [sharp](https://github.com/lovell/sharp), see https://github.com/lovell/sharp/issues/3395

Strapi uses sharp v0.31.1, this bug is fixed from up to 0.31.3. Until Strapi fixes the dependecy, you can f.e. override the version by adding 
`"resolutions": {
    "sharp": "0.31.3"
  },`

to your package.json (if you are using yarn).

**Comment by cooperwfloyd** [comment]
excellent find @JWandscheer, confirming that the following upgrade to the latest version of sharp solved the issue that I'm seeing. will open a corresponding PR to upgrade sharp 👍

npm:
```
"overrides": {
   "sharp": "^0.32.0"
 }
```

yarn:
```
"resolutions": {
   "sharp": "^0.32.0"
 }
```
